src/averageAges.js

```diff
@@ -38,6 +60,25 @@ function calculateMenAverageAge(people, century) {
  */
 function calculateWomenAverageAge(people, withChildren) {
   // write code here
+  const isWoman = people.filter((person) => person.sex === 'f');
+  const isWomanWithChildren = isWoman.filter((woman) =>
+    people.find((person) => person.mother === woman.name)
+  );
+
+  const sumAgeAllWoman = isWoman.reduce(
+    (sum, person) => sum + (person.died - person.born),
+    0
+  );
+  const averageAgeAllWoman = sumAgeAllWoman / isWoman.length;
+
+  const sumAgeWomanWithChildren = isWomanWithChildren.reduce(
+    (sum, person) => sum + (person.died - person.born),
+    0
+  );
+  const averageAgeWomanWithChildren
+  = sumAgeWomanWithChildren / isWomanWithChildren.length;
```

Comment:
The '=' sign should be on the same line as the variable declaration to maintain code consistency.